compiler/noirc_evaluator/ssa_test/src/main.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs::File;
+use std::io::Write;
+use std::io::Read;
+use std::path::Path;
+use flate2::read::GzDecoder;
+use::noirc_evaluator::acir_instruction_builder::{
+    InstructionArtifacts, VariableType, Variable
+};
+use clap::Parser;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long, default_value = "../../../../../barretenberg/cpp/src/barretenberg/acir_formal_proofs/artifacts/")]
+    dir: String,
+}
+
+/// Decompresses gzipped data into a byte vector
+fn ungzip(compressed_data: Vec<u8>) -> Vec<u8> {
+    let mut decompressed_data: Vec<u8> = Vec::new();
+    let mut decoder = GzDecoder::new(&compressed_data[..]);
+    decoder.read_to_end(&mut decompressed_data).unwrap();
+    return decompressed_data;
+}
+
+/// Saves byte data to a file at the specified path
+fn save_to_file(data: &[u8], filename: &str) -> Result<(), std::io::Error> {
+    let path = Path::new(filename);
+    let mut file = File::create(path)?;
+    file.write_all(data)?;
+    Ok(())
+}
+
+/// Saves instruction artifacts to files in the artifacts directory
+/// Prints the formatted SSA for each artifact and saves the decompressed ACIR
+fn save_artifacts(all_artifacts: Vec<InstructionArtifacts>, dir: &str) {
+    for artifacts in all_artifacts.iter() {
+        println!("{}", artifacts.formatted_ssa);
```

Comment:
Perhaps you can print a header of the artefact between the SSA to make it easier to recognise which one it is?